### PR TITLE
[tests] init

### PR DESCRIPTION
**Pull request by TomasVotruba**
### Todo

- [x] cover with test https://github.com/eventigo/eventigo-web/issues/20

### CODE REVIEW

**Comment by TomasVotruba** [comment]
I got only couple of minutes in the morning. Please check it if this covers the case in issue.
Gotta go now.

Thanks!

**Review by jakubzaba — APPROVED** [review]

File: tests/Modules/Newsletter/Model/NewsletterServiceTest.php [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    protected function setUp()
+    {
+        $container = (new ContainerFactory)->create();
+        $this->newsletterService = $container->getByType(NewsletterService::class);
+        $this->linkGenerator = $container->getByType(LinkGenerator::class);
+    }
+
+    public function testRedirectLinks()
+    {
+        $this->assertSame('https://eventigo.cz/redirect/?url=someUrl', $this->linkGenerator->link('Front:Redirect:', ['someUrl']));
```
Comment (filipsuk) [inline tests/Modules/Newsletter/Model/NewsletterServiceTest.php]: Minor detail: shouldn't we use base url from config?

**Review by filipsuk — CHANGES_REQUESTED** [review]

File: tests/Modules/Newsletter/Model/NewsletterServiceTest.php (hunk 1 above)
Comment (TomasVotruba) [reply tests/Modules/Newsletter/Model/NewsletterServiceTest.php]: You're right. That would make test change proof.

Comment (TomasVotruba) [reply tests/Modules/Newsletter/Model/NewsletterServiceTest.php]: Fixed!

**Comment by filipsuk** [comment]
@TomasVotruba is travis setup to run the tests? Does not seem like it to me. And also, if I run it on local, the test fails with :

```
There was 1 error:

1) App\Modules\Newsletter\Model\NewsletterServiceTest::testPrepareLinks
Error: Cannot access private property App\Modules\Newsletter\Model\NewsletterService::$context

/var/www/app/tests/Modules/Newsletter/Model/NewsletterServiceTest.php:47

--

There was 1 failure:

1) App\Modules\Newsletter\Model\NewsletterServiceTest::testRedirectLinks
Failed asserting that two strings are identical.
--- Expected
+++ Actual
@@ @@
-'https://eventigo.local/redirect/?url=someUrl'
+'http:///redirect/?url=someUrl'

/var/www/app/tests/Modules/Newsletter/Model/NewsletterServiceTest.php:39
```

However I am really looking forward to our first test 👍 🎉

**Comment by TomasVotruba** [comment]
> @TomasVotruba is travis setup to run the tests? Does not seem like it to me. 

Ha! Good catch. Not really, fixed :+1: 

**Travis is passing now.** 


Only Scrutinizer is bugged, could you please remove it from [services in Settings](https://github.com/eventigo/eventigo-web/settings/installations)?

Also, I'd switch to [more reliable Coveralls](https://www.tomasvotruba.cz/blog/2017/06/12/how-to-require-minimal-code-coverage-for-github-pull-requests-with-coveralls/). Could you [add the repository in there](https://coveralls.io/repos/new)?

**Comment by filipsuk** [comment]
Great! I removed scrutinizer from settings, hopefully its gone. Could not add coveralls, their repo sync is kinda buggy, try to remind me later. Thanks for this!

**Review by filipsuk — APPROVED** [review]

**Comment by TomasVotruba** [comment]
Thank you!

**Comment by TomasVotruba** [comment]
@filipsuk Could you try adding to Coveralls today? It might be in better mood now :)

**Comment by filipsuk** [comment]
@TomasVotruba ok repo is added to Coveralls, let me know if there is anything else I need to setup :)

**Comment by TomasVotruba** [comment]
@filipsuk Thanks a lot :+1:

**Pull request merged**